django/contrib/postgres/indexes.py

```diff
@@ -1,8 +1,33 @@
 from __future__ import unicode_literals
 
-from django.db.models import Index
+from django.db.models.indexes import Index
 
-__all__ = ['GinIndex']
+__all__ = ['BrinIndex', 'GinIndex']
+
+
+class BrinIndex(Index):
+    suffix = 'brin'
+
+    def __init__(self, *args, **kwargs):
+        self.pages_per_range = kwargs.pop('pages_per_range', 128)
+        return super(BrinIndex, self).__init__(*args, **kwargs)
+
+    def __eq__(self, other):
+        return super(BrinIndex, self).__eq__(other) and self.pages_per_range == other.pages_per_range
+
+    def __repr__(self):
+        return "<%(name)s: %(field)s, pages_per_range=%(pages_per_range)s>" % {
+            'name': self.__class__.__name__,
+            'fields': self.fields,
+            'pages_per_range': self.pages_per_range,
+        }
+
+    def create_sql(self, model, schema_editor):
+        using = ' USING brin'
+        if self.pages_per_range is not None:
+            using += " WITH (pages_per_range = {})".format(self.pages_per_range)
+        sql = super(BrinIndex, self).create_sql(model, schema_editor, using=' USING brin')
```

Comment:
Yes. MySQL/MariaDB includes a lot of options after the ``USING``-clause. Oddly enough, there's no where-clause in the syntax. It's possible in all other three backends to limit the index to a subset of the table. I was tempted to suggest renaming extra to where, but because this is not implemented in MySQL/MariaDB, I won't.

 Oracle (for me, this is easier to read than the the official documentation: http://ss64.com/ora/index_c.html) also has some options before tablespace. I suppose Oracle users use all those options, and thus either may specify the raw SQL in ``create_sql``, or not include it in the django-code at all. 

SQLite is rather simple, and apart from the where-clause, the syntax is rather plain.

PostgreSQL of course has the with-clause, and at the end, the where-clause.

So, instead of naming the variable ``with``, the name ``specials`` (or ``options`') may be better? Anyway,  

This discussion brings to mind that the current implementation of ``GinIndex`` is incomplete, because the options ``fastupdate`` and ``gin_pending_list_limit`` are not available. I don't know if this falls into the category "not every database abstraction should be supported by Django" as the code is already in master.